src/Endpoints/Keys.php

```diff
@@ -33,10 +33,32 @@ public function __construct(Http $http, $key = null, $description = null, $actio
         parent::__construct($http);
     }
 
+    protected function newInstance(array $attributes): self
+    {
+        $key = new self(
+            $this->http,
+            $attributes['key'],
+            $attributes['description'],
+            $attributes['actions'],
+            $attributes['indexes'],
+        );
+        if ($attributes['expiresAt']) {
+            $key->expiresAt = date_create_from_format('Y-m-d\TH:i:s\Z', $attributes['expiresAt']);
+        }
+        if ($attributes['createdAt']) {
+            $key->createdAt = date_create_from_format('Y-m-d\TH:i:s.vu\Z', $attributes['createdAt']);
+        }
+        if ($attributes['updatedAt']) {
+            $key->updatedAt = date_create_from_format('Y-m-d\TH:i:s.vu\Z', $attributes['updatedAt']);
+        }
```

Comment:
Yes, we can but it's not the behavior we want I think. The goal of not putting it in the `__construct` is that you can directly give it a `DateTime` instead of a `string`.
And this is a good question because we also can remove this if forest and put it in the `function all()` instead. But I think this `newInstance` method is more useful to translate a `json` document to a `Key` object.
What do you think?

Reply:
Indeed, we never use `new Key()` in the class because this is not its purpose. But if you are a user and you want to create a Key you do like:
```java
        Key key = new Key();
        key.setIndexes(new String[] {"*"});
        key.setActions(new String[] {"*"});
        key.setExpiresAt(null);

        client.createKey(key);
```
You can see more examples in the `code-sample.yml` file.
And for the `newInstance` examples I don't, I only used it here. But if you talk about `fill()` function (the old `newInstance()` function that I renamed) I used it a lot to fill the `Key` with a `json` response from `Meilisearch`.
Not sure is more clear?

Reply:
```
    public function create(array $options = []): self
```

[See here](https://github.com/meilisearch/meilisearch-php/blob/e7076f4bb2cb79494312729c4f07a1c4dcf4e1eb/src/Endpoints/Keys.php#L138)

Is this expecting an instance of `Keys` or an option object?

Reply:
The both can work but the field `expiredAt` should be `DateTime` in all case.

Reply:
I understand my mistake! LGTM !

Reply:
Shouldn't `DateTimeImmutable::createFromFormat()` be used instead of `date_create_from_format` ?